mission_control/src/mission_control.cpp

```diff
@@ -137,17 +142,39 @@ void MissionControlNode::reportHeartbeat(const ros::TimerEvent& timer)
 
 void MissionControlNode::executeMissionT(const ros::TimerEvent& timer)
 {
-  NodeStatus missionStatus = m_mission_map[currentMissionId]->getStatus();
-
-  if (missionStatus == BT::NodeStatus::IDLE || missionStatus == BT::NodeStatus::RUNNING)
+  BT::NodeStatus behaviorStatus;
+  switch (missionState)
   {
-    missionStatus = m_mission_map[currentMissionId]->Continue();
+    case mission_control::ReportExecuteMissionState::PAUSED:
+      behaviorStatus = m_mission_map[currentMissionId]->Continue();
+      break;
+    case mission_control::ReportExecuteMissionState::EXECUTING:
+      behaviorStatus = m_mission_map[currentMissionId]->Continue();
+      break;
+    case mission_control::ReportExecuteMissionState::COMPLETE:
+      executeMissionTimer.stop();
+      behaviorStatus = BT::NodeStatus::SUCCESS;
+      break;
+    case mission_control::ReportExecuteMissionState::ABORTING:
+      abortMission(currentMissionId);
+      behaviorStatus = BT::NodeStatus::FAILURE;
+      break;
   }
```

Comment:
@patapufate +1 to conceptually separating mission state from behavior tree state. What I do think is that this should occur within the `Mission` abstraction, not here. Code here should at most react to state changes as reported by the mission.

Reply:
Done

Reply:
Hmm, I still see `switch` statements mapping states and even a `missionState` member in the `mission_control` main node class.